// @theo4u/ng-alert/esm5/lib/alert.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { CloseType } from './interfaces';
var NgAlertComponent = /** @class */ (function () {
    function NgAlertComponent() {
        this.messageChange = new EventEmitter();
        this.closeType = CloseType.TIMES;
        this.closeTypes = CloseType;
    }
    Object.defineProperty(NgAlertComponent.prototype, "message", {
        get: /**
         * @return {?}
         */
        function () {
            return this.rawMessage;
        },
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this.rawMessage = value;
            this.messageChange.emit(this.rawMessage);
        },
        enumerable: true,
        configurable: true
    });
    NgAlertComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ng-alert',
                    template: "<div *ngIf=\"message\" class=\"alert alert-{{message.type}}\" [ngClass]=\"{padding:closeType===closeTypes.TIMES}\">\n  <h5 *ngIf=\"message.title\">{{message.title}}</h5>\n  <span [innerHTML]=\"message.message\">\n  </span>\n\n  <button *ngFor=\"let button of message.buttons\" class=\"{{button.css}}\" (click)=\"button.action?button.action():true\">\n    {{button.label}}\n  </button>\n\n  <button (click)=\"message=null\" *ngIf=\"dismissable && closeType===closeTypes.TIMES\" type=\"button\" class=\"close\" data-dismiss=\"alert\"\n                                                                                                                 aria-label=\"Close\">\n    <span aria-hidden=\"true\">&times;</span>\n  </button>\n  <p align=\"center\" *ngIf=\"dismissable && closeType===closeTypes.NORMAL\">\n  <a (click)=\"message=null\">close</a>\n  </p>\n\n</div>\n"
                }] }
    ];
    NgAlertComponent.propDecorators = {
        dismissable: [{ type: Input }],
        messageChange: [{ type: Output }],
        closeType: [{ type: Input }],
        message: [{ type: Input }]
    };
    return NgAlertComponent;
}());
export { NgAlertComponent };
if (false) {
    /** @type {?} */
    NgAlertComponent.prototype.dismissable;
    /** @type {?} */
    NgAlertComponent.prototype.rawMessage;
    /** @type {?} */
    NgAlertComponent.prototype.messageChange;
    /** @type {?} */
    NgAlertComponent.prototype.closeType;
    /** @type {?} */
    NgAlertComponent.prototype.closeTypes;
}
