// @theo4u/ng-alert/fesm2015/theo4u-ng-alert.js
import { Injectable, Component, Input, Output, EventEmitter, NgModule, defineInjectable } from '@angular/core';
import { Subject } from 'rxjs';
import { CommonModule } from '@angular/common';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
/** @enum {string} */
const MessageType = {
    success: 'success',
    error: 'error',
    info: 'info',
    warning: 'warning',
};
/** @enum {number} */
const CloseType = {
    TIMES: 0, NORMAL: 1,
};
CloseType[CloseType.TIMES] = 'TIMES';
CloseType[CloseType.NORMAL] = 'NORMAL';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class NgAlertService {
    constructor() {
        this._messageSource = new Subject();
    }
    /**
     * this returns our alert message source
     * @return {?}
     */
    getSource() {
        return this._messageSource;
    }
    /**
     * push our message to our alert
     * @param {?} message type IMessage to be pushed to our listeners/subscribers
     * @return {?}
     */
    push(message) {
        this._messageSource.next(message);
    }
}
NgAlertService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
NgAlertService.ctorParameters = () => [];
/** @nocollapse */ NgAlertService.ngInjectableDef = defineInjectable({ factory: function NgAlertService_Factory() { return new NgAlertService(); }, token: NgAlertService, providedIn: "root" });

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class NgAlertComponent {
    constructor() {
        this.messageChange = new EventEmitter();
        this.closeType = CloseType.TIMES;
        this.closeTypes = CloseType;
    }
    /**
     * @return {?}
     */
    get message() {
        return this.rawMessage;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set message(value) {
        this.rawMessage = value;
        this.messageChange.emit(this.rawMessage);
    }
}
NgAlertComponent.decorators = [
    { type: Component, args: [{
                selector: 'ng-alert',
                template: "<div *ngIf=\"message\" class=\"alert alert-{{message.type}}\" [ngClass]=\"{padding:closeType===closeTypes.TIMES}\">\n  <h5 *ngIf=\"message.title\">{{message.title}}</h5>\n  <span [innerHTML]=\"message.message\">\n  </span>\n\n  <button *ngFor=\"let button of message.buttons\" class=\"{{button.css}}\" (click)=\"button.action?button.action():true\">\n    {{button.label}}\n  </button>\n\n  <button (click)=\"message=null\" *ngIf=\"dismissable && closeType===closeTypes.TIMES\" type=\"button\" class=\"close\" data-dismiss=\"alert\"\n                                                                                                                 aria-label=\"Close\">\n    <span aria-hidden=\"true\">&times;</span>\n  </button>\n  <p align=\"center\" *ngIf=\"dismissable && closeType===closeTypes.NORMAL\">\n  <a (click)=\"message=null\">close</a>\n  </p>\n\n</div>\n"
            }] }
];
NgAlertComponent.propDecorators = {
    dismissable: [{ type: Input }],
    messageChange: [{ type: Output }],
    closeType: [{ type: Input }],
    message: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class NgAlertModule {
}
NgAlertModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule
                ],
                declarations: [NgAlertComponent],
                exports: [NgAlertComponent]
            },] }
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

export { MessageType, CloseType, NgAlertService, NgAlertComponent, NgAlertModule };
