// @theo4u/ng-alert/bundles/theo4u-ng-alert.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('rxjs'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('@theo4u/ng-alert', ['exports', '@angular/core', 'rxjs', '@angular/common'], factory) :
    (factory((global.theo4u = global.theo4u || {}, global.theo4u['ng-alert'] = {}),global.ng.core,global.rxjs,global.ng.common));
}(this, (function (exports,i0,rxjs,common) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    /** @enum {string} */
    var MessageType = {
        success: 'success',
        error: 'error',
        info: 'info',
        warning: 'warning',
    };
    /** @enum {number} */
    var CloseType = {
        TIMES: 0, NORMAL: 1,
    };
    CloseType[CloseType.TIMES] = 'TIMES';
    CloseType[CloseType.NORMAL] = 'NORMAL';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var NgAlertService = /** @class */ (function () {
        function NgAlertService() {
            this._messageSource = new rxjs.Subject();
        }
        /**
         * this returns our alert message source
         */
        /**
         * this returns our alert message source
         * @return {?}
         */
        NgAlertService.prototype.getSource = /**
         * this returns our alert message source
         * @return {?}
         */
            function () {
                return this._messageSource;
            };
        /**
         * push our message to our alert
         * @param message type IMessage to be pushed to our listeners/subscribers
         */
        /**
         * push our message to our alert
         * @param {?} message type IMessage to be pushed to our listeners/subscribers
         * @return {?}
         */
        NgAlertService.prototype.push = /**
         * push our message to our alert
         * @param {?} message type IMessage to be pushed to our listeners/subscribers
         * @return {?}
         */
            function (message) {
                this._messageSource.next(message);
            };
        NgAlertService.decorators = [
            { type: i0.Injectable, args: [{
                        providedIn: 'root'
                    },] }
        ];
        /** @nocollapse */
        NgAlertService.ctorParameters = function () { return []; };
        /** @nocollapse */ NgAlertService.ngInjectableDef = i0.defineInjectable({ factory: function NgAlertService_Factory() { return new NgAlertService(); }, token: NgAlertService, providedIn: "root" });
        return NgAlertService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var NgAlertComponent = /** @class */ (function () {
        function NgAlertComponent() {
            this.messageChange = new i0.EventEmitter();
            this.closeType = CloseType.TIMES;
            this.closeTypes = CloseType;
        }
        Object.defineProperty(NgAlertComponent.prototype, "message", {
            get: /**
             * @return {?}
             */ function () {
                return this.rawMessage;
            },
            set: /**
             * @param {?} value
             * @return {?}
             */ function (value) {
                this.rawMessage = value;
                this.messageChange.emit(this.rawMessage);
            },
            enumerable: true,
            configurable: true
        });
        NgAlertComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'ng-alert',
                        template: "<div *ngIf=\"message\" class=\"alert alert-{{message.type}}\" [ngClass]=\"{padding:closeType===closeTypes.TIMES}\">\n  <h5 *ngIf=\"message.title\">{{message.title}}</h5>\n  <span [innerHTML]=\"message.message\">\n  </span>\n\n  <button *ngFor=\"let button of message.buttons\" class=\"{{button.css}}\" (click)=\"button.action?button.action():true\">\n    {{button.label}}\n  </button>\n\n  <button (click)=\"message=null\" *ngIf=\"dismissable && closeType===closeTypes.TIMES\" type=\"button\" class=\"close\" data-dismiss=\"alert\"\n                                                                                                                 aria-label=\"Close\">\n    <span aria-hidden=\"true\">&times;</span>\n  </button>\n  <p align=\"center\" *ngIf=\"dismissable && closeType===closeTypes.NORMAL\">\n  <a (click)=\"message=null\">close</a>\n  </p>\n\n</div>\n"
                    }] }
        ];
        NgAlertComponent.propDecorators = {
            dismissable: [{ type: i0.Input }],
            messageChange: [{ type: i0.Output }],
            closeType: [{ type: i0.Input }],
            message: [{ type: i0.Input }]
        };
        return NgAlertComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var NgAlertModule = /** @class */ (function () {
        function NgAlertModule() {
        }
        NgAlertModule.decorators = [
            { type: i0.NgModule, args: [{
                        imports: [
                            common.CommonModule
                        ],
                        declarations: [NgAlertComponent],
                        exports: [NgAlertComponent]
                    },] }
        ];
        return NgAlertModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */

    exports.MessageType = MessageType;
    exports.CloseType = CloseType;
    exports.NgAlertService = NgAlertService;
    exports.NgAlertComponent = NgAlertComponent;
    exports.NgAlertModule = NgAlertModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
